### ISSUE TITLE
Size parameter in "newImageFromIconName" should have type "IconSize"

### ISSUE THREAD

**Issue by lscrd** [StefanSalewski/gintro#6]
Not really an issue, rather an improvement.

In "newImageFromIconName", the size parameter should have type "IconSize" (which is an enum), not  "int". Indeed, allowed sizes are those defined by the enum, not actual icon sizes.

**Comment by StefanSalewski** [comment]
Thanks for reporting.

That bug is indeed based on a problem in the gobject-introspection gir files. There we have

/usr/share/gir-1.0/Gtk-3.0.gir

<type name="gint" c:type="GtkIconSize"/

So ctype is correct, but name is gint==int32. Generally our bindings use name. I have to investigate if this is a problem for all enum parameters. Special logic may be needed for this case then.

**Comment by StefanSalewski** [comment]
That seems to be a bug in the C header files for GtkIconSize. Unfortunately IconSize is used by a few procs, so manually fixing is some work. And I have still no idea for automatic fix.

You may try something like IconSize.menu.ord as parameters for now.

**Comment by lscrd** [comment]
Yes, this is what I have done (except that it was for a dialog, not a menu). I didn't considerer this to be a real problem, but it may cause errors if we are not careful (but it’s better to look at the GTK3 documentation in any case).

**Comment by StefanSalewski** [comment]
I just got a reply from GTK mailing list pointing to this:

https://bugzilla.gnome.org/show_bug.cgi?id=601425

So it is not really a bug in GTK, but a decision. For Nim making that parameter a plain int with named constants would be not a foolproof solution unfortunately, as people may try to pass real sizes like 16, 32, 64...

**Comment by lscrd** [comment]
So, changing the parameter to an enum is not an option. And changing the enum values to be constants would just be a small improvement, avoiding to use "ord" to get the value. Not the most important change to do, for sure.
